RestaurantsNearMe/Presentation/ContentView.swift

```diff
@@ -7,30 +7,41 @@
 
 import SwiftUI
 import Combine
+import CoreLocation
+import Networking
 
 struct ContentView: View {
-  @ObservedObject var splashScreenViewModel: SplashScreenViewModel
+  @ObservedObject var splashScreenViewModel: SplashScreenViewModel<YelpRequest>
```

Comment:
Does this feel odd to you? Why should the view care about `YelpRequest`?

Reply:
Ya it does,

I'm not super familiar on best practices regarding separation of layers between the `Network` and `UI`. pre `SwiftUI` would this go into the `ViewController`?

Ultimately I want the API call to be driven by UI State, i.e when the splashScreen is finished loading and the user accepts the location agreement, we immediately call the API. Given that I was trying to avoid existentials, this seemed reasonable at the time. Tho does kinda of look like a smell on reflection

Reply:
I guess this bleeds over into our protocol discussion above. In theory, `SplashScreenViewModel` should keep its internals (e.g. which request API it will be using) to itself, and not leak it to the view.